test/games/strategy/triplea/delegate/StratBombTest.java

```diff
@@ -0,0 +1,116 @@
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.ITestDelegateBridge;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.RouteScripted;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.random.IRandomSource;
+import games.strategy.engine.random.ScriptedRandomSource;
+import games.strategy.test.TestUtil;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.TripleAUnit;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.player.ITripleAPlayer;
+import games.strategy.triplea.xml.LoadGameUtil;
+
+public class StratBombTest {
+  private GameData m_data;
+
+  @Before
+  public void setUp() throws Exception {
+    m_data = LoadGameUtil.loadTestGame("ww2global40_2nd_edition_test.xml");
+  }
+
+  private ITestDelegateBridge getDelegateBridge(final PlayerID player) {
+    return GameDataTestUtil.getDelegateBridge(player, m_data);
+  }
+
+
+  @Test
+  public void testBombingRaid2targets() {
+    final Territory germany = m_data.getMap().getTerritory("Germany");
+    final Territory uk = m_data.getMap().getTerritory("United Kingdom");
+    final PlayerID germans = GameDataTestUtil.germans(m_data);
+    final PlayerID british = GameDataTestUtil.british(m_data);
+    // add a unit
+    final Unit stratBomber = GameDataTestUtil.bomber(m_data).create(british);
+    final Unit tacBomber1 = GameDataTestUtil.tacBomber(m_data).create(british);
+    final Unit tacBomber2 = GameDataTestUtil.tacBomber(m_data).create(british);
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(stratBomber)));
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(tacBomber1)));
+    m_data.performChange(ChangeFactory.addUnits(uk, Collections.singleton(tacBomber2)));
+    final BattleTracker tracker = new BattleTracker();
+    List<Unit> attackers = Collections.emptyList();
+    attackers.add(stratBomber);
+    attackers.add(tacBomber1);
+    attackers.add(tacBomber2);
+    HashMap<Unit, HashSet<Unit>> targets = null;
+   
+
+//    final Collection<Unit> enemyTargets = Match.getMatches(uk.getUnits().getMatches(new CompositeMatchAnd<>(Matches.enemyUnit(germans, m_data),              Matches.UnitIsAtMaxDamageOrNotCanBeDamaged(uk).invert(), Matches.unitIsBeingTransported().invert())), Matches.UnitIsLegalBombingTargetBy(stratBomber));
+
+    for( final Unit target : germany.getUnits().getUnits() ) {
```

Comment:
There's a NPE in this code once the  IllegalAccessException to the EmptyList is fixed (by switching it to an `new ArrayList<>`). Test code should be simple, a for loop with switch statement is just suspect, not a place to be debugging  a NPE.

 At the very basic unwind the for loop and list out each iteration explicitly, you should know the test data ordering. I still don't think that is really the right thing to do when the root problem is excessive test code complexity, that is the thing to be fixed. 

3 mini tests can serve as good control cases. Regardless, these are suggestions. Please take them constructively. In fact, it is really preferred for there to be no conversation in a CR other than commits to the code that address the comments to the extent you agree. Better to focus the energy to the code, anything said or done in a CR is going to be of no benefit to future developers or players.